### PR TITLE
Chore/2020 q1 update

### PR DESCRIPTION
**Pull request by jrconlin**
* includes actix 2.0 update
* fixes test_chan integration test script

NOTE: the new integration test currently fails if using pytest. The app hangs while trying to `recv` on the websocket. I'm using a simple python test that does the same operations and checks and works. I will revisit the pytest problem in a future issue.

### CODE REVIEW

**Comment by tublitzed** [comment]
Closes #77

File: test_chan/test_chan/__main__.py [hunk 1]
```diff
@@ -0,0 +1,227 @@
+import time
```
Comment (tublitzed) [inline test_chan/test_chan/__main__.py]: will this get picked up by circle, or do we need to manually run it?

Comment (jrconlin) [reply test_chan/test_chan/__main__.py]: It should get picked up automatically by circle.

File: test_chan/test_chan/test_chan.py [hunk 2]
```diff
@@ -148,36 +151,62 @@ def test_full_exchange(self):
...
             # Try reading, ignoring an already closed socket.
             bob.recv()
-        except websocket._exceptions.WebSocketConnectionClosedException:
+        except WebSocketConnectionClosedException:
             pass
+
         assert bob.is_closed(), "Receiver did not close"
 
     def test_bad_connection(self):
+        import pdb; pdb.set_trace()
```
Comment (tublitzed) [inline test_chan/test_chan/test_chan.py]: ✂️ ?

File: channelserver/src/logging.rs [hunk 3]
```diff
@@ -106,6 +106,7 @@ impl Display for LogMessage {
     }
 }
 
+/*
```
Comment (fzzzy) [inline channelserver/src/logging.rs]: Can this code just be removed?

File: channelserver/src/main.rs [hunk 4]
```diff
@@ -1,357 +1,162 @@
...
-use actix::Arbiter;
-use actix_web::middleware;
-use actix_web::server::HttpServer;
-use actix_web::{fs, http, ws, App, AsyncResponder, Error, HttpMessage, HttpRequest, HttpResponse};
-use futures::Future;
+use futures::future::Future;
+use serde_json::Value;
+use slog::{debug, error, warn};
+
+use actix::*;
```
Comment (fzzzy) [inline channelserver/src/main.rs]: This smells funny to me. I'd rather see the explicit imports.

File: channelserver/src/main.rs [hunk 5]
```diff
@@ -1,357 +1,162 @@
...
-    );
-
-    // Cannot apply headers here.
     ws::start(
-        req,
         session::WsChannelSession {
             id: 0,
             hb: Instant::now(),
-            expiry: Instant::now() + Duration::from_secs(req.state().connection_lifespan),
+            // initial_connect: initial_connect,
```
Comment (fzzzy) [inline channelserver/src/main.rs]: How come we aren't using initial_connect any more? We don't need it?

File: channelserver/src/main.rs [hunk 6]
```diff
@@ -1,357 +1,162 @@
...
 }
 
-fn heartbeat(req: &HttpRequest<session::WsChannelSessionState>) -> Result<HttpResponse, Error> {
+pub fn heartbeat(_req: HttpRequest) -> impl Future<Output = Result<HttpResponse, Error>> {
     // if there's more to check, add it here.
-    let body = json!({"status": "ok", "version": env!("CARGO_PKG_VERSION")});
-    Ok(HttpResponse::Ok()
+    let mut checklist = HashMap::new();
+    checklist.insert(
+        "version",
```
Comment (fzzzy) [inline channelserver/src/main.rs]: 👍

File: channelserver/src/server.rs [hunk 7]
```diff
@@ -1,21 +1,24 @@
...
 //! And manages available channels. Peers send messages to other peers in same
-//! channel through `ChannelServer`.
-
+//! channels through `ChannelServer`.
 use std::collections::{hash_map::Entry, HashMap};
 use std::fmt;
 use std::time::Instant;
 
-use actix::prelude::{Actor, Context, Handler, Recipient};
+use actix::prelude::*;
```
Comment (fzzzy) [inline channelserver/src/server.rs]: Let's be explicit here too.

File: channelserver/src/session.rs [hunk 8]
```diff
@@ -1,114 +1,168 @@
...
 use cadence::{Counted, StatsdClient};
 use ipnet::IpNet;
-use maxminddb;
+use slog::{debug, error, info};
 
-use channelid::ChannelID;
-use logging;
-use meta::SenderData;
-use server;
+use actix::*;
```
Comment (fzzzy) [inline channelserver/src/session.rs]: Explicit here.

File: channelserver/src/main.rs (hunk 5 above)
Comment (jrconlin) [reply channelserver/src/main.rs]: Ah, good catch! Had to fix up something test related and forgot to revert this.

File: channelserver/src/server.rs (hunk 7 above)
Comment (jrconlin) [reply channelserver/src/server.rs]: (shakes fist at cut and paste from examples)

**Pull request merged**